### PR TITLE
Check if source_from_index is in iib_no_ocp_label_allow_list

### PR DESCRIPTION
**Pull request by lipoja**
I wrongly assumed that ISV will use overwrite index token and set the check if image is on allow list only for the target (like we do this for bootstraping). However now I know it is not true and this use case slipped my hands. We have to also tests the source index image because the target can be newly created empty index image in our IIB repo. 

[CLOUDDST-20869]

### CODE REVIEW

**Review by yashvardhannanavati — APPROVED** [review]

File: iib/workers/tasks/build_merge_index_image.py [hunk 1]
```diff
@@ -128,9 +128,10 @@ def _add_bundles_missing_in_source(
             missing_bundles.append(bundle)
             missing_bundle_paths.append(bundle['bundlePath'])
 
-    if ignore_bundle_ocp_version and target_index is not None:
+    if ignore_bundle_ocp_version:
+        target_index_tmp = '' if target_index is None else target_index
         allow_no_ocp_version = any(
-            target_index.startswith(index)
+            target_index_tmp.startswith(index) or source_from_index.startswith(index)
             for index in get_worker_config()['iib_no_ocp_label_allow_list']
```
Comment (xDaile) [inline iib/workers/tasks/build_merge_index_image.py:135]: Does this change take into account that "random_text".startswith('') will result in True?

**Review by xDaile — APPROVED** [review]

File: iib/workers/tasks/build_merge_index_image.py (hunk 1 above)
Comment (lipoja) [reply iib/workers/tasks/build_merge_index_image.py:135]: well, it did not . However I added config validation to ensure nobody sets empty string in the allow list.
Please re-review

Comment (xDaile) [reply iib/workers/tasks/build_merge_index_image.py:135]: My bad, when I was thinking about this, I switched the parameters (I thought the target index was checked in `startswith()` method). However, this check is useful in my opinion.

**Review by xDaile — APPROVED** [review]
